string[] sentences
string sentence_id
string[] noun_id
string[] noun_str
